### PR TITLE
TY-4136 Add config for changing max_queue_size

### PR DESCRIPTION
**Pull request by tc-szhu**

### CODE REVIEW

File: src/eredis_sub_client.erl [hunk 1]
```diff
@@ -57,7 +57,8 @@ init([Host, Port, Password, ReconnectSleep, MaxQueueSize, QueueBehaviour]) ->
     case connect(State) of
         {ok, NewState} ->
             ok = inet:setopts(NewState#state.socket, [{active, once}]),
-            {ok, NewState};
+            {ok, New_TimerRef} = timer:send_after(1000 * 60, timeout),
+            {ok, NewState#state{interval=60,tref=New_TimerRef}};
```
Comment (tc-szhu) [inline src/eredis_sub_client.erl:61]: At this stage, redis driver is not ready, so it use a default value.

File: src/eredis_sub_client.erl [hunk 2]
```diff
@@ -148,18 +149,23 @@ handle_info({tcp, _Socket, Bs}, State) ->
...
-            end;
-        false ->
-            {noreply, NewState}
+    case NewState#state.max_queue_size of
+        infinity ->
+            {noreply, NewState};
+        MaxQueueSize ->
+            case queue:len(NewState#state.msg_queue) > MaxQueueSize of
+                true ->
+                    case State#state.queue_behaviour of
```
Comment (TT-lwang) [inline src/eredis_sub_client.erl:159]: better to have log if exceeding max_queue_size

Comment (tc-szhu) [reply src/eredis_sub_client.erl:159]: fixed.

**Review by TT-lwang — APPROVED** [review]
LGTM

**Comment by tc-szhu** [comment]
This change is only for xmpp, so we will not merge it to master branch.

**Pull request closed without merging**